southbound/device-factory/device-factory-core/src/main/java/org/eclipse/sensinact/gateway/southbound/device/factory/ValueType.java

```diff
@@ -26,38 +26,69 @@
 import java.util.stream.Stream;
 
 import org.eclipse.sensinact.gateway.southbound.device.factory.dto.DeviceMappingOptionsDTO;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public enum ValueType {
     /**
      * Represent available types
      */
-    AS_IS("any", null),
-    STRING("string", (v, o) -> String.valueOf(v)),
-    INT("int", (v, o) -> {
+    AS_IS("any", null, null), STRING("string", (v, o) -> String.valueOf(v), String.class), CHAR("char", (v, o) -> {
+        if (v instanceof Character) {
+            return (Character) v;
+        } else if (v instanceof CharSequence) {
+            final String str = v.toString();
```

Comment:
We don't need to use a String here. Cast to CharSequence and then check for length `0`.